pkg/compiler/codegen.go

```diff
@@ -488,8 +488,16 @@ func (c *codegen) Visit(node ast.Node) ast.Visitor {
 			return nil
 		}
 
+		args := n.Args
+		isAppCall := isAppCall(n.Fun)
+		// When using APPCALL, script hash is a parameter taken from script itself
+		// not from a hash. Script hash should be emitted after APPCALL.
```

Comment:
It took me a while to parse this sentence, the most important thing here is that APPCALL parameter is a part of instruction rather than an element on the stack, maybe writing it this way will make things easier to understand.